pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -310,13 +311,15 @@ protected int calculateNumOfMessageToRead(int currentTotalAvailablePermits) {
                     }
                     topic.getBrokerService().executor().schedule(() -> readMoreEntries(), MESSAGE_RATE_BACKOFF_MS,
                             TimeUnit.MILLISECONDS);
-                    return -1;
+                    return Pair.of(-1, -1L);
                 } else {
-                    // if dispatch-rate is in msg then read only msg according to available permit
-                    long availablePermitsOnMsg = topicRateLimiter.getAvailableDispatchRateLimitOnMsg();
-                    if (availablePermitsOnMsg > 0) {
-                        messagesToRead = Math.min(messagesToRead, (int) availablePermitsOnMsg);
-                    }
+                    Pair<Integer, Long> calculateResult = computeReadLimits(messagesToRead,
+                            (int) topicRateLimiter.getAvailableDispatchRateLimitOnMsg(),
+                            topicRateLimiter.getAvailableDispatchRateLimitOnByte(), bytesToRead);
```

Comment:
You have not changed the order of arguments here (after changing the order in the function definition). It should be

```java
 computeReadLimits(
    messagesToRead,
    (int) topicRateLimiter.getAvailableDispatchRateLimitOnMsg(),
    bytesToRead,
    topicRateLimiter.getAvailableDispatchRateLimitOnByte());
```

Please fix at other places too where this function is called.

Reply:
done, could you please review again? if don't have any other problem, please approve this pr then we will merge it .